server/src/main/java/org/opensearch/cluster/metadata/ResourceLimitGroup.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class to define the ResourceLimitGroup schema
+ * {
+ *     "analytics":{
+ *              "jvm": 0.4,
```

Comment:
This looks inconsistent with the class attributes below. `jvm` should be within resourceLimits?

Reply:
It is not inconsistent the resourceLimits in the code are modeled using map but the `toXContent` method will just add the attributes of the resource limits plain and simple.